packages/twenty-front/src/modules/workflow/workflow-actions/components/WorkflowEditActionFormRecordDelete.tsx

```diff
@@ -0,0 +1,170 @@
+
+type WorkflowEditActionFormRecordDeleteProps = {
+  action: WorkflowRecordDeleteAction;
+  actionOptions:
+    | {
+        readonly: true;
+      }
+    | {
+        readonly?: false;
+        onActionUpdate: (action: WorkflowRecordDeleteAction) => void;
+      };
+};
+
+type DeleteRecordFormData = {
+  objectName: string;
+  objectRecordId: string;
+};
+
+export const WorkflowEditActionFormRecordDelete = ({
+  action,
+  actionOptions,
+}: WorkflowEditActionFormRecordDeleteProps) => {
+  const theme = useTheme();
+  const { getIcon } = useIcons();
+
+  const { activeObjectMetadataItems } = useFilteredObjectMetadataItems();
+
+  const availableMetadata: Array<SelectOption<string>> =
+    activeObjectMetadataItems.map((item) => ({
+      Icon: getIcon(item.icon),
+      label: item.labelPlural,
+      value: item.nameSingular,
+    }));
+
+  const [formData, setFormData] = useState<DeleteRecordFormData>({
+    objectName: action.settings.input.objectName,
+    objectRecordId: action.settings.input.objectRecordId,
+  });
+  const isFormDisabled = actionOptions.readonly;
+
+  const handleFieldChange = (
+    fieldName: keyof DeleteRecordFormData,
+    updatedValue: JsonValue,
+  ) => {
+    const newFormData: DeleteRecordFormData = {
+      ...formData,
+      [fieldName]: updatedValue,
+    };
+
+    setFormData(newFormData);
+
+    saveAction(newFormData);
+  };
+
+  useEffect(() => {
+    setFormData({
+      objectName: action.settings.input.objectName,
+      objectRecordId: action.settings.input.objectRecordId,
+    });
+  }, [action.settings.input]);
+
+  const selectedObjectMetadataItemNameSingular = formData.objectName;
+
+  const selectedObjectMetadataItem = activeObjectMetadataItems.find(
+    (item) => item.nameSingular === selectedObjectMetadataItemNameSingular,
+  );
+  if (!isDefined(selectedObjectMetadataItem)) {
+    throw new Error('Should have found the metadata item');
+  }
+
+  const saveAction = useDebouncedCallback(
+    async (formData: DeleteRecordFormData) => {
+      if (actionOptions.readonly === true) {
+        return;
+      }
+
+      const {
+        objectName: updatedObjectName,
+        objectRecordId: updatedObjectRecordId,
+      } = formData;
+
+      actionOptions.onActionUpdate({
+        ...action,
+        settings: {
+          ...action.settings,
+          input: {
+            type: 'DELETE',
+            objectName: updatedObjectName,
+            objectRecordId: updatedObjectRecordId ?? '',
+          },
+        },
+      });
+    },
+    1_000,
+  );
+
+  useEffect(() => {
+    return () => {
+      saveAction.flush();
+    };
+  }, [saveAction]);
+
+  const headerTitle = isDefined(action.name) ? action.name : `Delete Record`;
+
+  return (
+    <WorkflowEditGenericFormBase
+      onTitleChange={(newName: string) => {
+        if (actionOptions.readonly === true) {
+          return;
+        }
+
+        actionOptions.onActionUpdate({
+          ...action,
+          name: newName,
+        });
```

Comment:
I wonder why we don't just add this `readonly` check directly in the onActionUpdate